File: Java/167-TwoSumII-InputArrayIsSorted.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers.length < 2) return new int[2];
        int start = 0, end = numbers.length-1;
        while (start < end){
            if (numbers[start] + numbers[end] == target)
                break;
            else if (numbers[start] + numbers[end] > target)
                end--;
            else
                start++;
        }
        return new int[]{start+1, end+1};
    }
}

File: cpp/167-TwoSumII-InputArrayIsSorted.cpp
//[2,7,11,15] target = 9
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        if (nums.size() < 2)
            return res;
        int i1 = 0, i2 = nums.size() - 1;
        while (i1 < i2){
            if (nums[i1] + nums[i2] == target){
                res.push_back(i1+1);
                res.push_back(i2+1);
                return res;
            }else if (nums[i1] + nums[i2] > target){
                i2--;
            }else{
                i1++;
            }
        }
        return res;
    }
};
